garak/detectors/custom.py

```diff
@@ -0,0 +1,313 @@
+        "case_sensitive": False,
+    }
+    
+    active = False
+    tags = []
+    
+    def __init__(self, config_root=_config):
+        """Initialize RegexDetector from config.
+        
+        Args:
+            config_root: Configuration root object containing detector configuration
+        
+        Raises:
+            ValueError: If no patterns provided or invalid regex pattern
+        """
+        super().__init__(config_root=config_root)
+        
+        # Normalize to list
+        if isinstance(self.patterns, str):
+            self.patterns = [self.patterns]
+        
+        # Validate match_type
+        self.match_type = str(self.match_type).lower()
+        if self.match_type not in ("any", "all"):
+            raise ValueError(f"match_type must be 'any' or 'all', got '{self.match_type}'")
+        
+        # Compile regex patterns
+        flags = 0 if self.case_sensitive else re.IGNORECASE
+        try:
+            self.compiled_patterns = [
+                re.compile(pattern, flags) for pattern in self.patterns
+            ]
+        except re.error as e:
+            raise ValueError(f"Invalid regex pattern: {e}") from e
+        
+        # Update description
+        self.description = (
+            f"Regex detector matching {self.match_type} of {len(self.patterns)} pattern(s)"
+        )
+        
+        logging.info(
+            "RegexDetector initialized with %d patterns (match_type=%s, case_sensitive=%s)",
+            len(self.patterns), self.match_type, self.case_sensitive
+        )
+    
+    def detect(self, attempt: garak.attempt.Attempt) -> List[float]:
+        """Detect if outputs match regex patterns.
+        
+        Returns 1.0 if match found (vulnerability), 0.0 otherwise.
+        
+        Args:
+            attempt: Attempt object containing outputs to check
+            
+        Returns:
+            List of scores (0.0 or 1.0) for each output
+        """
+        results = []
+        for output in attempt.outputs:
+            output_text = output.text if output else None
+            if output_text is None:
+                results.append(None)
+                continue
+            
+            if not self.compiled_patterns:
+                results.append(0.0) # default to safe if no patterns are provided
+                continue
+
+            # Check patterns based on match_type
+            if self.match_type == "any":
+                # Match if ANY pattern is found
+                matched = any(
+                    pattern.search(output_text) for pattern in self.compiled_patterns
+                )
+            elif self.match_type == "all":
+                # Match only if ALL patterns are found
+                matched = all(
+                    pattern.search(output_text) for pattern in self.compiled_patterns
+                )
+            else:
+                raise ValueError(f"Invalid match_type: {self.match_type}")
```

Comment:
Please log and skip, rather than risking aborting the run. Failing detector can safely return `[None] * input_length`. Would be best if we accepted detectors returning flat `None` on overall detector failure but I'm not sure that's possible yet.